postgres/client.sls

```diff
@@ -1,17 +1,18 @@
-{% from "postgres/map.jinja" import postgres with context %}
+{%- from "postgres/map.jinja" import postgres with context -%}
+
+{%- set pkgs = [] %}
+{%- for pkg in (postgres.pkg_client, postgres.pkg_libpq_dev) %}
+  {%- if pkg %}
+    {%- do pkgs.append(pkg) %}
+  {%- endif %}
+{%- endfor -%}
 
-{% if postgres.use_upstream_repo %}
 include:
   - postgres.upstream
-{% endif %}
-
-install-postgresql-client:
-  pkg.installed:
-    - name: {{ postgres.pkg_client }}
-    - refresh: {{ postgres.use_upstream_repo }}
 
-{% if postgres.pkg_libpq_dev %}
-install-postgres-libpq-dev:
+postgresql-client:
   pkg.installed:
-    - name: {{ postgres.pkg_libpq_dev }}
-{% endif %}
+    - pkgs: {{ pkgs }}
+    - refresh: {{ 'pkg_repo' in postgres }}
+    - require:
```

Comment:
Same as in my previous comment, this requirement should either be conditional or, better yet, a `require_in` set in the `install-postgresql-repo` in the `postgres/upstream.sls` file, don't you agree?